File: ios/DataAnalyzer/ScanTemplate.swift
import Foundation

// MARK: - ScanTemplate Model
class ScanTemplate: Codable {
    
    enum STLabelType: Int, Codable {
        case text = 1
        case qrCode = 2
        case barcode = 3
    }

    enum STLValueRegexType: String, Codable {
        case none = ""
        case alphaNumeric = "alphanumeric-string"
        case macAddress = "mac-address"
        case serialNumber = "serial-number"
        case ipAddress = "ip-address"
        case imei = "imei"
        case anyValue = "any"

        // Support alternate formats manually if needed
        static func from(rawValue: String) -> STLValueRegexType {
            switch rawValue.lowercased() {
            case "alphanumeric-string": return .alphaNumeric
            case "mac-address", "xx xx xx xx xx xx", "xx:xx:xx:xx:xx:xx": return .macAddress
            case "serial-number": return .serialNumber
            case "ip-address", "xxx.xxx.xxx.xxx": return .ipAddress
            case "imei": return .imei
            case "any": return .anyValue
            default: return .none
            }
        }
    }

    class STLabel: Codable {
        var identifier: String = ""
        var type: STLabelType = .text
        var ordinal: Int = 0
        var prefix: String = ""
        var regexType: STLValueRegexType = .none
    }

    var name: String = ""
    var labels: [STLabel] = []
    var image: String = ""
}

File: android/src/main/java/com/deviceonboarder/labelScan/model/ScanTemplate.kt
package com.deviceonboarder.labelScan.model

import com.google.gson.annotations.SerializedName

class ScanTemplate {

    enum class STLabelType(val labelType: Int) {
        TEXT(1),
        QRCODE(2),
        BARCODE(3)
    }

    enum class STLValueRegexType(val regexType: String) {
        NONE(""),
        @SerializedName("alphanumeric-string")
        ALPHA_NUMERIC("alphanumeric-string"),
        @SerializedName("mac-address", alternate = ["XX XX XX XX XX XX", "XX:XX:XX:XX:XX:XX"])
        MAC_ADDRESS("mac-address"),
        @SerializedName("serial-number")
        SERIAL_NUMBER("serial-number"),
        @SerializedName("ip-address", alternate = ["XXX.XXX.XXX.XXX"])
        IP_ADDRESS("ip-address"),
        @SerializedName("imei")
        IMEI("imei"),
        @SerializedName("any")
        ANY_VALUE("any"),
    }

    class STLabel {
        lateinit var identifier: String
        lateinit var type: STLabelType
        var ordinal: Int = 0
        lateinit var prefix: String
        lateinit var regexType: STLValueRegexType
    }

    lateinit var name: String
    var labels: ArrayList<STLabel> = arrayListOf()
    lateinit var image: String
}
